lib/note-revisions/index.tsx

```diff
@@ -3,70 +3,179 @@ import { connect } from 'react-redux';
+};
+
+export class NoteRevisions extends Component<Props, State> {
   static displayName = 'NoteRevisions';
 
+  state: State = {
+    includeDeletedTags: true,
+  };
+
+  getTags() {
+    return this.props.noteCanonicalTags
+      .map((tagName) => {
+        const tagHash = tagHashOf(tagName);
+        return { name: tagName, deleted: !this.props.allTags.has(tagHash) };
+      })
+      .filter((tag) => {
+        return this.state.includeDeletedTags || !tag.deleted;
+      });
+  }
+
+  onAcceptRevision = () => {
+    const { noteId, note, revision, restoreRevision } = this.props;
+
+    if (!noteId || !note || !revision) {
+      return;
+    }
+
+    const noteEmailTags = note.tags.filter((tagName) => isEmailTag(tagName));
+    const revisionCanonicalTags = this.getTags().map(({ name }) => name);
+
+    restoreRevision(noteId, {
+      ...revision,
+      tags: [...noteEmailTags, ...revisionCanonicalTags],
+      systemTags: note.systemTags,
+    });
+  };
+
+  onSelectRevision = (revisionIndex: number) => {
+    const { noteId, revisions, openRevision } = this.props;
+
+    if (!noteId || !revisions) {
+      return;
+    }
+
+    const revision = [...revisions.keys()][revisionIndex];
+
+    openRevision(noteId, revision);
+  };
+
+  renderTags() {
+    const tags = this.getTags();
+
+    return (
+      <div className="tags">
+        {tags.map(({ name, deleted }) => (
+          <TagChip
+            key={name}
+            tagName={name}
+            interactive={false}
+            deleted={deleted}
+          />
+        ))}
+      </div>
+    );
+  }
+
   render() {
-    const { note, noteId, tags, 'aria-hidden': ariaHidden } = this.props;
+    const {
+      'aria-hidden': ariaHidden,
+      cancelRevision,
+      noteId,
+      revision,
+      revisionIndex,
+      revisions,
+    } = this.props;
+
+    if (!noteId || !revisions) {
+      return null;
+    }
 
     return (
-      <div aria-hidden={ariaHidden} className="note-revisions">
-        <NotePreview noteId={noteId} note={note} />
-        <div className="tags">
-          {tags.map(({ name, deleted }) => (
-            <TagChip
-              key={name}
-              tagName={name}
-              interactive={false}
-              deleted={deleted}
-            />
-          ))}
+      <div className="note-revisions">
+        <div aria-hidden={ariaHidden} className="note-revision-preview">
+          <NotePreview noteId={noteId} note={revision} />
+          {this.renderTags()}
         </div>
+        <RevisionSelector
+          onAccept={this.onAcceptRevision}
+          onCancel={cancelRevision}
+          onSelect={this.onSelectRevision}
+          onToggleRestoreDeletedTags={(toggled) =>
+            this.setState({ includeDeletedTags: toggled })
+          }
+          restoreDeletedTags={this.state.includeDeletedTags}
+          revision={revision}
+          revisionsSize={revisions.size}
+          selectedIndex={revisionIndex}
+        />
```

Comment:
I extracted all the required props for disconnecting the `RevisionSelector` component from Redux, this way `NoteRevision` handles the logic related to revision management and `RevisionSelector` only handles the UI logic.